ApollosProject/apollos-shovel
# Crossroads Algolia Indexing

### About

This PR creates an algolia index for Crossroads Kids Club and Crossroads TV and adds the indexing step to both the backfill dag and the sync dag.

I largely copied the code from `dags/algolia_multitenant/algolia_content_dag.py`, and made changes necessary for crossroads. The following changes are relevant:

- We're using Crossroads' Algolia instance instead of the cluster's. Crossroads pays for a fancy algolia account and I see us wanting to use some of those features in the future. 
- I'm mapping the description field to the summary field in algolia. Crossroads content only has a description, no summary. To do so, I had to remove the HTML markup from the description field using the new `HTMLParser` class.

I went ahead and took this opportunity to reorganize the DAGs a bit too. For both kids club and tv, we had only been running the `create_categories` step during a backfill. There's no reason we can't run the `create_categories` step during a sync too, so I removed some unnecessary logic from each of the DAGs.

### Testing
1. Run the sync DAG for kids club and crossroads tv. Verify that the algolia indexing step completes.
2. Check out the results in algolia's dashboard (I'm happy to give you the credentials for this account) and verify everything looks good.

### Screenshots
||Backfill|Sync|
|---|---|---|
|Kids' Club|<img width="380" alt="Screen Shot 2022-02-07 at 3 07 36 PM" src="https://user-images.githubusercontent.com/34075293/152863768-7032f2ed-f462-4997-b9e7-8f696798904d.png">|<img width="618" alt="Screen Shot 2022-02-07 at 2 58 06 PM" src="https://user-images.githubusercontent.com/34075293/152863749-adb94d2c-1bdd-4f7b-a4f6-c18eb86b2330.png">|
|Crossroads TV|<img width="408" alt="Screen Shot 2022-02-09 at 10 03 19 AM" src="https://user-images.githubusercontent.com/34075293/153228218-72dab367-0f7f-46b8-bb57-e24260ee28bf.png">|<img width="659" alt="Screen Shot 2022-02-09 at 10 02 59 AM" src="https://user-images.githubusercontent.com/34075293/153228259-53f16b7e-e29e-4e1b-b303-fb5091e8ab9e.png">|
